src/SurveyService.Data/QuestionRepository.cs

```diff
@@ -11,6 +13,25 @@ public class QuestionRepository : IQuestionRepository
 {
   private readonly IDataProvider _provider;
 
+  private IQueryable<DbQuestion> CreateGetPredicates(
+    GetQuestionFilter filter,
+    IQueryable<DbQuestion> dbQuestions)
```

Comment:
you get question by Id -> it can be question (not list) or null in result

Reply:
It wouldn't be a list as id is unique, but can be null. Check added